cmdutil/service/standard.go

```diff
@@ -101,7 +102,13 @@ func New(appConfig interface{}, ofs ...OptionFunc) *Standard {
 // Add adds cmdutil.Servers to be managed.
 func (s *Standard) Add(svs ...cmdutil.Server) {
 	for _, sv := range svs {
-		s.g.Add(sv.Run, sv.Stop)
+		sv := sv
+		runWithPanicReport := func() error {
```

Comment:
Why not do this wrapping in `New()` by utilizing the `options`? That feels a little more in-line with the existing options pattern.

Reply:
The crux is related to this (from [here](https://go.dev/blog/defer-panic-and-recover)):

> Panic is a built-in function that stops the ordinary flow of control and begins panicking. When the function F calls panic, execution of F stops, any deferred functions in F are executed normally, and then F returns to its caller. To the caller, F then behaves like a call to panic. The process continues up the stack until all functions in the current goroutine have returned, at which point the program crashes. 

In my tests, this is the only way the `defer`'d functions get actually called. The program crash happens when [this goroutine](https://github.com/oklog/run/blob/master/group.go#L37-L39) panics where `a.execute` is the call to `sv.Run`. I tried several other variations only this one worked. Happy to be proven wrong though.

Reply:
okay, that's fine by me. i see what you mean